### PR TITLE
Fix autofocus feed CIDR regex

### PR DESCRIPTION
**Pull request by GuyAfik**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/45892

## Description
This PR will fix the classification of the CIDR and will implement a different logic for indicator types classification 
of the following integrations: 
1. FeedAutofocus
2. FeedAutofocusDaily

In addition more tests will be added to those two integration due to lack of test coverage. 

In addition this PR will also update the URL-regex to support more IP types such as:
1) Hexa-decimal IPs
2) octa based IPs
3) IPv6 support
4) Integer IPs

## Screenshots

now the samples like X.X.X.X/X/path/morepath will be classified as URL type indicators instead of CIDR

image

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [x] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by GuyAfik** [comment]
failed unit-tests for both FeedAutofocus and FeedAutofocusDaily integration - https://app.circleci.com/pipelines/github/demisto/content/139643/workflows/55960b52-4625-423e-a2b0-d7f8d9367bbf/jobs/471908

File: Packs/AutoFocus/Integrations/FeedAutofocus/FeedAutofocus.py [hunk 1]
```diff
@@ -326,55 +326,31 @@ def create_indicators_from_single_sample_response(single_sample):
...
-        ip_type = self.get_ip_type(indicator)
-        if ip_type:
-            # catch URLs of type X.X.X.X/path/url or X.X.X.X:portNum/path/url
-            if '/' in indicator and (ip_type not in [FeedIndicatorType.IPv6CIDR, FeedIndicatorType.CIDR]):
+        if ip_type := FeedIndicatorType.ip_to_indicator_type(indicator):
+            # catch URLs of type X.X.X.X/path/url or X.X.X.X:portNum/path/url or X.X.X.X/cidrNumber/path/url
+            # or X.X.X.X:portNum/path
+            if indicator.count('/') > 1 or (
+                ip_type in (FeedIndicatorType.IP, FeedIndicatorType.IPv6) and indicator.count('/') == 1
+            ):
```
Comment (ShahafBenYakir) [inline Packs/AutoFocus/Integrations/FeedAutofocus/FeedAutofocus.py:345]: Not sure I understand this `if`. If the function doesn't return the right type to means there is a bug in the regex no?

File: Packs/AutoFocus/ReleaseNotes/2_0_8.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
+
+#### Integrations
+##### AutoFocus Feed
+- Improved the classification of the indicator types.
```
Comment (ShahafBenYakir) [inline Packs/AutoFocus/ReleaseNotes/2_0_8.md:4]:
```suggestion
- Improved the classification of URL indicator types.
```

File: Packs/AutoFocus/ReleaseNotes/2_0_8.md [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+- Updated the Docker image to: *demisto/python3:3.10.1.25933*.
+##### AutoFocus Daily Feed
+- Improved the classification of the indicator types.
```
Comment (ShahafBenYakir) [inline Packs/AutoFocus/ReleaseNotes/2_0_8.md:7]:
```suggestion
- Improved the classification of URL indicator types.
```

File: Packs/AutoFocus/Integrations/FeedAutofocus/FeedAutofocus.py (hunk 1 above)
Comment (GuyAfik) [reply Packs/AutoFocus/Integrations/FeedAutofocus/FeedAutofocus.py:345]: right now the regex ```urlRegex``` that is defined in the common-server-python is defined only for dns based URLs, for example ```https://github.com/demisto/content/pull/17204``` and not for ```IP``` based URLs such as ```1.1.1.1/path```.  I am not sure if the ```urlRegex``` in common server python should fit to ```IP/path``` so I decided not to change it until the review (I also saw it is being widely used in many places in our code - so don't want break stuff).

**Review by ShahafBenYakir — APPROVED** [review]

**Pull request merged**